src/enumerable.cr

```diff
@@ -306,19 +299,12 @@ module Enumerable(T)
   #     #=> 3
   #     #=> 4
   #
-  def in_groups_of(size : Int, filled_up_with : U = nil)
+  def in_groups_of(size : Int, filled_up_with : U = nil, &block)
     raise ArgumentError.new("size must be positive") if size <= 0
-    ary = Array(T | U).new(size, filled_up_with)
-
-    # TODO: this consumes the enumerable twice, fix
-    count = self.size
 
-    each_with_index do |e, i|
-      ary[i % size] = e
-      if i % size == size - 1 || i == count - 1
-        yield ary
-        ary = Array(T | U).new(size, filled_up_with)
-      end
+    each_slice(size) do |slice|
+      group = slice.size < size ? slice + [filled_up_with] : slice
```

Comment:
Ah, does it mean `assert { [1, 2, 3, 4].in_groups_of(3).should eq([[1, 2, 3], [4, nil, nil]]) }`?
It sounds so reasonable to me, but current behavior does not work for it. Would I fix the behavior in this PR?